osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SystemEventServiceTest.java

```diff
@@ -0,0 +1,151 @@
+        new Builder().withOrganisationIdentification("org-id").build();
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    verifyNoInteractions(this.correlationIdProviderService);
+    verify(this.osgpRequestMessageSender, never())
+        .send(
+            any(RequestMessage.class),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            any(MessageMetadata.class));
+  }
+
+  @Test
+  void verifyMaxValueReachedEvent() {
+    final DlmsDevice device =
+        new DlmsDeviceBuilder()
+            .withDeviceIdentification("device-1")
+            .withInvocationCounter(this.invocationCounterEventThreshold)
+            .build();
+
+    final MessageMetadata messageMetadata =
+        new Builder()
+            .withOrganisationIdentification("org-id")
+            .withDomain("domain")
+            .withDomainVersion("1.0")
+            .build();
+
+    when(this.correlationIdProviderService.getCorrelationId(
+            messageMetadata.getOrganisationIdentification(), device.getDeviceIdentification()))
+        .thenReturn("corr-id");
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    final ArgumentCaptor<MessageMetadata> messageMetadataCaptor =
+        ArgumentCaptor.forClass(MessageMetadata.class);
+    final ArgumentCaptor<RequestMessage> requestMessageCaptor =
+        ArgumentCaptor.forClass(RequestMessage.class);
+
+    verify(this.osgpRequestMessageSender)
+        .send(
+            requestMessageCaptor.capture(),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            messageMetadataCaptor.capture());
+
+    final RequestMessage requestMessage = requestMessageCaptor.getValue();
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(requestMessage.getCorrelationUid()).isEqualTo("corr-id");
+    assertThat(requestMessage.getOrganisationIdentification()).isEqualTo("no-organisation");
+    assertThat(requestMessage.getIpAddress()).isNull();
+    assertThat(requestMessage.getRequest()).isInstanceOf(SystemEventDto.class);
+
+    final SystemEventDto systemEventDto = (SystemEventDto) requestMessage.getRequest();
+    assertThat(systemEventDto.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(systemEventDto.getSystemEventType())
+        .isEqualTo(SystemEventTypeDto.INVOCATION_COUNTER_THRESHOLD_REACHED);
+    assertThat(systemEventDto.getTimestamp()).isNotNull();
+
+    final MessageMetadata metadata = messageMetadataCaptor.getValue();
+    assertThat(metadata.getDeviceIdentification()).isEqualTo(device.getDeviceIdentification());
+    assertThat(metadata.getCorrelationUid()).isEqualTo("corr-id");
+    assertThat(metadata.getOrganisationIdentification()).isEqualTo("no-organisation");
```

Comment:
Same comment as a few lines above.
The duplication of the assertions with the `RequestMessage` and the `MessageMetadata` does not look nice, but maybe we should leave that as it is for now. The intended way of refactoring the message handling would probably end up loosing the `RequestMessage` as a wrapper, and keep only the actual payload (here that would be the `SystemEventDto`) and the `MessageMetadata`, but we are not there yet.

Reply:
Changed, getOrganisationIdentification copied from metadata